### PR TITLE
fix for partial deploy with blue-green, LMCROSSITXSADEPLOY-2154

### PR DESCRIPTION
**Pull request by ikasarov**
taking selected modules '-m module-name' into consideration when renaming applications at start of blue-green strategy deply

### CODE REVIEW

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetectApplicationsToRenameStep.java [hunk 1]
```diff
@@ -49,13 +51,28 @@ protected StepPhase executeStep(ProcessContext context) {
         return StepPhase.DONE;
     }
 
-    private List<String> computeOldAppsToRename(Set<String> appsToProcess) {
-        return appsToProcess.stream()
-                            .filter(appName -> !BlueGreenApplicationNameSuffix.isSuffixContainedIn(appName))
-                            .collect(Collectors.toList());
+    private List<String> computeOldAppsToRename(DeployedMta deployedMta, List<String> selectedModulesList) {        
+        Set<String> selectedModulesSet = new TreeSet<>();
```
Comment (radito3) [inline multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetectApplicationsToRenameStep.java:55]: I don't understand the need for this variable.
Just pass the list (and remove the `List` suffix from its name) to the other method

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetectApplicationsToRenameStep.java [hunk 2]
```diff
@@ -49,13 +51,28 @@ protected StepPhase executeStep(ProcessContext context) {
...
+
+        return deployedMta.getApplications()
+                          .stream()
+                          .filter(app -> !BlueGreenApplicationNameSuffix.isSuffixContainedIn(app.getName()))
+                          .filter(app -> moduleIsSelectedForDeployment(app.getModuleName(), selectedModulesSet))
+                          .map(DeployedMtaApplication::getName)
+                          .collect(Collectors.toList());
+    }
+
+    private boolean moduleIsSelectedForDeployment(String module, Set<String> selectedModules) {
```
Comment (radito3) [inline multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetectApplicationsToRenameStep.java:66]: Wouldn't `isModuleSelectedForDeployment` be better?
And rename the first parameter to `moduleName`

**Review by radito3 — CHANGES_REQUESTED** [review]

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetectApplicationsToRenameStep.java (hunk 1 above)
Comment (ikasarov) [reply multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetectApplicationsToRenameStep.java:55]: the set will have better performance when later making the contains check

Comment (ikasarov) [reply multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetectApplicationsToRenameStep.java:55]: will remove in service of code clarity, performance isn't really a concern here

**Pull request merged**